pkg/globalnet/controllers/gateway_monitor.go

```diff
@@ -335,6 +335,10 @@ func (g *gatewayMonitor) createGlobalNetMarkingChain() error {
 		return errors.Wrapf(err, "error creating iptables chain %s", constants.SmGlobalnetMarkChain)
 	}
 
+	if err := iptables.AwaitChain(g.ipt, "nat", constants.SmGlobalnetMarkChain); err != nil {
```

Comment:
I guess this error can happen with any IPT chain creation and not just with 'SUBMARINER-GN-MARK', , shouldn't we handle this also for the rest of the cases?

Reply:
Yes it can happen with other chains as well, but i want to handle SUBMARINER-GN-MARK alone in this PR. We can push separate PR to address other chains.

Reply:
Perhaps we should add this check in `NewChain` in the IPT wrapper.

Reply:
Yes, we can do that too. I'm currently going through the logs in Noam's setup and trying to fix issues based on the logs.